docs/az-aro-python-development.md

```diff
@@ -0,0 +1,102 @@
+# How-to `az aro` development
+
+ARO-RP distributed the CLI command `az aro` to control the cluster. 
+Currently the `aro` subcommand is merged in the 
+[azure-cli](https://github.com/azure/azure-cli) code base and 
+is present as a command.
+
+Currently, there are two codebases for `aro` subcommand:
+
+* [upstream](https://github.com/Azure/ARO-RP/tree/master/python/az/aro) in main ARO-RP repo - which is an `az` extension
+* [downstream][https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/aro] - which is an `az` command
+
+Command is distributed alongside `az` and is automaticaly present in Azure cloud shell. Extension is installed as user action.
+
+Currently code have to be in sync for upstream and downstream.
+
+## Set up dev environment `az` for `aro` command
+
+Python virtual environment is required:
+
+```
+python3 -m venv env
```

Comment:
FWIW the virtualenv this provides is not the same as the OS package virtualenv that we use elsewhere. There are some subtle differences.